Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -139,14 +139,20 @@ public class Klaviyo: NSObject  {
     }
     
     /**
-     handlePush: Extracts tracking information from received push notification and sends the data to Klaviyo for push-tracking
-     analystics.
+     handlePush: Sends the push payload back to Klaviyo as an opened event
+        if the message originated from Klaviyo, as determined by presence of _k tracking parameters
      
      - Parameter userInfo: NSDictionary containing the push notification text & metadata
      */
     public func handlePush(userInfo: NSDictionary) {
-        if let _ = userInfo["_k"] as? NSDictionary {
-            trackEvent(eventName: KLPersonOpenedPush, properties: userInfo)
+        // Test it originated from klaviyo by checking presence of _k tracking parameters
+        if let body = userInfo["body"] as? NSDictionary, let _ = body["_k"] as? NSDictionary {
+            //But we still want to send the full payload, plus the push token
+            _ = environment.analytics.store.state.sink{ completion in
```

Comment:
I need to read up on this composed architecture pattern some more, maybe this isn't the proper way to access current state?